src/NServiceBus.Core/Recoverability/DefaultRecoverabilityPolicy.cs

```diff
@@ -4,58 +4,102 @@ namespace NServiceBus
     using Logging;
     using Transport;
 
-    class DefaultRecoverabilityPolicy : IRecoverabilityPolicy
+
+    /// <summary>
+    /// The default recoverability policy.
+    /// </summary>
+    public static class DefaultRecoverabilityPolicy
     {
-        public DefaultRecoverabilityPolicy(bool immediateRetriesEnabled, bool delayedRetriesEnabled, int maxImmediateRetries, SecondLevelRetryPolicy secondLevelRetryPolicy)
+        /// <summary>
+        /// Invokes the default recovery policy.
+        /// </summary>
+        /// <param name="config">The recoverability configuration.</param>
+        /// <param name="errorContext">The error context.</param>
+        /// <returns>The recoverability action.</returns>
+        public static RecoverabilityAction Invoke(RecoverabilityConfig config, ErrorContext errorContext)
         {
-            this.immediateRetriesEnabled = immediateRetriesEnabled;
-            this.delayedRetriesEnabled = delayedRetriesEnabled;
-            this.maxImmediateRetries = maxImmediateRetries;
-            this.secondLevelRetryPolicy = secondLevelRetryPolicy;
+            if (errorContext.Exception is MessageDeserializationException)
+            {
+                return RecoverabilityAction.MoveToError(config.Failed.ErrorQueue);
+            }
+
+            if (errorContext.NumberOfFailedImmediateDeliveryAttempts <= config.Immediate.MaxNumberOfRetries)
+            {
+                return RecoverabilityAction.ImmediateRetry();
+            }
+
+            Logger.InfoFormat("Giving up First Level Retries for message '{0}'.", errorContext.Message.MessageId);
```

Comment:
we could wrap that in an `if(config.Immediate.MaxNumberOfRetries > 0)` but not sure it would add much value.